ambry-commons/src/main/java/com.github.ambry.commons/BlobId.java

```diff
@@ -24,28 +25,116 @@
 import java.util.UUID;
 import org.apache.commons.codec.binary.Base64;
 
+import static com.github.ambry.clustermap.Datacenter.*;
+import static com.github.ambry.commons.Account.*;
+import static com.github.ambry.commons.Container.*;
+
 
 /**
- * BlobId uniquely identifies a stored blob as well as the Partition in which the blob is stored.
+ * <p>
+ *   BlobId uniquely identifies a stored blob. A blobId is a reference that is returned back to a caller when
+ *   posting a blob, and later will be required to fetch the blob.
+ * </p>
+ * <p>
+ *   There are two versions of format for blob de/serialization. Version 1, which includes {@code partitionId}
+ *   of the blob. The {@code partitionId} is the {@link com.github.ambry.clustermap.Partition} to which this
+ *   blob is assigned.
+ * </p>
+ * <pre>
+ * +---------------------------------------------+
+ * | version | partitionId | uuidSize | uuid     |
+ * | (short) | (n bytes)   | (short)  | (n bytes)|
+ * +---------------------------------------------+
+ * </pre>
+ * <p>
+ *   Version 2, which includes{@code flag}, {@code accountId}, {@code containerId}, {@code datacenterId}, and
+ *   {@code partitionId} of the blob. {@code flag} is a single byte that carries the meta information for this
+ *   blobId. The {@code datacenterId} is the id of the datacenter where this blob was originally posted (not
+ *   through replication). The {@code accountId} is the id of the {@link Account} the blob belongs to. The
+ *   {@code containerId} is the {@link Container} the blob belongs to. The {@code partitionId} is the
+ *   {@link com.github.ambry.clustermap.Partition} to which this blob is assigned.
+ * </p>
+ * <pre>
+ * +---------+-------+--------------+-----------+-------------+-------------+----------+----------+
+ * | version | flag  | datacenterId | accountId | containerId | partitionId | uuidSize | uuid     |
+ * | (short) | (byte)| (byte)       | (short)   | (short)     | (n bytes)   | (short)  | (n bytes)|
+ * +---------+----------------------+-----------+-------------+-------------+----------+----------+
+ * </pre>
  */
 public class BlobId extends StoreKey {
-  private Short version = 1;
-  private static short Version_Size_In_Bytes = 2;
+  static final short BLOB_ID_V1 = 1;
```

Comment:
Tests heavily used it. Do you think it makes sense to make it `protected` and extend the `BlobId` class?